Bottom: 268987378831ae3f9a4504443767dc29c53f3c08
Top:    6b668a034e3daca16fa3bb2d434e03e194412577
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-07-23 16:15:58 +0100

Allow editing of ordinary text objects when element names are hidden

Closes-bug: lp-810678


---

diff --git a/src/search.c b/src/search.c
index 30fe883..60f6f46 100644
--- a/src/search.c
+++ b/src/search.c
@@ -1167,11 +1167,14 @@ SearchObjectByLocation (int Type,
       SearchBox = point_box (X, Y);
     }
 
-  if (TEST_FLAG (LOCKNAMESFLAG, PCB)
-      || TEST_FLAG (HIDENAMESFLAG, PCB))
+  if (TEST_FLAG (LOCKNAMESFLAG, PCB))
     {
       Type &= ~ (ELEMENTNAME_TYPE | TEXT_TYPE);
     }
+  if (TEST_FLAG (HIDENAMESFLAG, PCB))
+    {
+      Type &= ~ELEMENTNAME_TYPE;
+    }
   if (TEST_FLAG (ONLYNAMESFLAG, PCB))
     {
       Type &= (ELEMENTNAME_TYPE | TEXT_TYPE);
